### DISCUSSION TITLE
Server: Self-Hosting Cloud Storage

### DISCUSSION THREAD

**Discussion by Pantyhose-X** [stoatchat/discussions#499] [Feature Suggestions]
Allow server administrators to add Autumn Self-Hosting Cloud Storage
All files uploaded locally by users are stored in Self-Hosting Cloud Storage

**Comment by sussycatgirl** [comment]
Not sure what exactly you're proposing, but:

- If you mean we should allow users to use their own Nextcloud server to host their files, then no - This would make it trivial to obtain user's IP addresses. And if you're going to proxy those files, they might as well be hosted on Autumn directly. Though you could probably write a plugin that uploads your files to Nextcloud and sends a link to the file instead of attaching it via Autumn.
- If you want to substitute Autumn with Nextcloud in selfhosted environments, I can guarantee you the overhead of Autumn + Minio is negligible compared to that of Nextcloud.

**Reply by Pantyhose-X** [reply]
Users upload files all hosted on self-deployed Autumn + Minio
ip is indeed a problem.
